### PR TITLE
Added player position to map

### PR DESCRIPTION
**Pull request by tanis2000**
This adds the player position that the client sends in the request so that we can track that as well.

### CODE REVIEW

File: ui/player.json [hunk 1]
```diff
@@ -0,0 +1 @@
+{"features": [{"geometry": {"coordinates": [9.676348686218262, 45.69357681274414], "type": "Point"}, "id": "player", "properties": {"id": "player", "marker-color": "663399", "marker-size": "large", "marker-symbol": "pitch", "title": "You", "type": "player"}, "type": "Feature"}], "type": "FeatureCollection"}
```
Comment (bettse) [inline ui/player.json]: don't check this in...actually you should add the filename to the gitignore

File: ui/index.js [hunk 2]
```diff
@@ -34,6 +35,34 @@ realtime.on('update', function(updateEvent) {
   }
 });
 
+
+var player = L.realtime({url: 'player.json', type: 'json'}, {
+  interval: 10 * 1000,
+  style: function(feature) { return feature.properties; },
+  pointToLayer: simplestyle,
+
+  filter: function(feature) {
```
Comment (bettse) [inline ui/index.js]: Unless I missed where you added a filter ui element for player, you shouldn't have a filter method

File: ui/index.js [hunk 3]
```diff
@@ -34,6 +35,34 @@ realtime.on('update', function(updateEvent) {
...
+    var box = document.getElementById(feature.properties.type);
+    return box == null || box.checked;
+  },
+  onEachFeature: function (feature, layer) {
+    layer.bindPopup(feature.properties.title);
+  }
+}).addTo(map);
+
+var playerLoaded = false;
+player.on('update', function(updateEvent) {
```
Comment (bettse) [inline ui/index.js]: I think you can drop all of this, the other data points do a better fitBound IMHO

File: decode.py [hunk 4]
```diff
@@ -264,5 +282,7 @@ def response(context, flow):
         dump = geojson.dumps(fc, sort_keys=True)
         f = open('ui/get_map_objects.json', 'w')
         f.write(dump)
+#      elif (key == GET_PLAYER_PROFILE):
```
Comment (bettse) [inline decode.py]: drop these; I try to not check in commented out code

File: decode.py [hunk 5]
```diff
@@ -134,6 +134,24 @@ def request(context, flow):
     except:
       print("Missing Request API: %s" % name)
 
+    if (key == GET_MAP_OBJECTS):
+      features = []
+      props = {
+          "id": "player"
+      }
+      props["marker-symbol"] = "pitch"
```
Comment (bettse) [inline decode.py]: these can all be set in the initial dict declaration since they're static

File: decode.py [hunk 6]
```diff
@@ -134,6 +134,24 @@ def request(context, flow):
...
+      props["title"] = "You"
+      props["type"] = "player"
+      props["marker-size"] = "large"
+      props["marker-color"] = "663399"
+      p = Point((mor.PlayerLng, mor.PlayerLat))
+      f = Feature(geometry=p, id="player", properties=props)
+      features.append(f)
+      fc = FeatureCollection(features)
```
Comment (bettse) [inline decode.py]: you may not need a full blown feature collection, but it doens't hurt to plan for the future, so this seems good

File: ui/player.json (hunk 1 above)
Comment (tanis2000) [inline ui/player.json]: Forgot about it, will do!

File: ui/index.js (hunk 2 above)
Comment (tanis2000) [inline ui/index.js]: I forgot to add it in the HTML. I'm adding it there right now.

File: ui/index.js (hunk 3 above)
Comment (tanis2000) [inline ui/index.js]: I'm dropping the fitBound. The other is better indeed.

File: decode.py [hunk 7]
```diff
@@ -134,6 +134,23 @@ def request(context, flow):
     except:
       print("Missing Request API: %s" % name)
 
+    if (key == GET_MAP_OBJECTS):
+      features = []
+      props = {
+          "id": "player",
```
Comment (bettse) [inline decode.py]: For the filtering, you'll need a 'type' in the dict to match the id of the html input element

File: ui/index.js (hunk 2 above)
Comment (bettse) [inline ui/index.js]: given that, we could move this function down and give it a name and reference it like we do for pointToLayer, but that's just an idea and not an expectation for this PR.

**Comment by bettse** [comment]
👍

**Pull request merged**